File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minSoFar = prices[0];
        int ans = 0;

        for (int i = 1; i < prices.size(); i++) {
            int profit = prices[i] - minSoFar;
            if (profit > ans) {
                ans = profit;
            }

            minSoFar = min(prices[i], minSoFar);
        }

        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minSoFar = prices[0]
        ans = 0

        for i in range(1, len(prices)):
            profit = prices[i] - minSoFar
            if profit > ans:
                ans = profit

            minSoFar = min(prices[i], minSoFar)

        return ans
        